### PR TITLE
SemEval 2024 Task 2: Add practice_test and test data

### PR DESCRIPTION
**Pull request by leonweber**
Adds practice_test and test data, setting `label = ""` where it is now known.

### CODE REVIEW

File: bigbio/hub/hub_repos/sem_eval_2024_task_2/sem_eval_2024_task_2.py [hunk 1]
```diff
@@ -258,6 +284,16 @@ def _generate_examples(self, data_dir: Path, split: str, config) -> Tuple[int, D
                 else:
                     premise = f" Primary: {text_primary}"
 
-                yield id_, {"id": id_, "premise": premise, "hypothesis": raw_label_data[id_]["Statement"], "label": raw_label_data[id_]["Label"]}
+                yield id_, {"id": id_, "premise": premise, "hypothesis": raw_label_data[id_]["Statement"], "label": raw_label_data[id_].get("Label")}
         else:
             raise ValueError(f"Unknown schema {self.config.schema}")
+
+    def _load_split_file(self, data_dir_or_path, split):
```
Comment (mariosaenger) [inline bigbio/hub/hub_repos/sem_eval_2024_task_2/sem_eval_2024_task_2.py:291]: Add parameter and return type datatype hints

File: bigbio/hub/hub_repos/sem_eval_2024_task_2/sem_eval_2024_task_2.py [hunk 2]
```diff
@@ -169,35 +171,58 @@ def _info(self) -> datasets.DatasetInfo:
     def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
         """Returns SplitGenerators."""
 
-        urls = _URLS["train"]
-        data_dir = Path(dl_manager.download_and_extract(urls))
+        train_dev_dir = Path(dl_manager.download_and_extract(_URLS["train"]))
```
Comment (mariosaenger) [inline bigbio/hub/hub_repos/sem_eval_2024_task_2/sem_eval_2024_task_2.py:174]: My rewrite it as `paths = { split: dl_manager.download_and_extract[url] for split, url in _URLS.items()}` and then access the paths via dictionary lookups in the code below.

**Review by mariosaenger — CHANGES_REQUESTED** [review]
Thanks for updating the data set @leonweber!

Unfortunately, I got a FileNotFoundError when running the tests:

FileNotFoundError: [Errno 2] No such file or directory: '/home/mario/.cache/huggingface/datasets/downloads/extracted/70cc6884a1b9f1d45b0cba813e0649c690968a99ee024faab2524a9fd4780d76/train.json'

File: bigbio/hub/hub_repos/sem_eval_2024_task_2/sem_eval_2024_task_2.py (hunk 2 above)
Comment (leonweber) [reply bigbio/hub/hub_repos/sem_eval_2024_task_2/sem_eval_2024_task_2.py]: Should be fixed with 450a5494e6fe2a968bb5dfede34193c095c41784

File: bigbio/hub/hub_repos/sem_eval_2024_task_2/sem_eval_2024_task_2.py (hunk 1 above)
Comment (leonweber) [reply bigbio/hub/hub_repos/sem_eval_2024_task_2/sem_eval_2024_task_2.py]: Should be fixed with 450a5494e6fe2a968bb5dfede34193c095c41784

**Review by mariosaenger — APPROVED** [review]

**Pull request merged**